scipy/sparse/_compressed.py

```diff
@@ -1464,6 +1464,42 @@ def _broadcast_to(self, shape, copy=False):
             indices = np.tile(np.arange(N), len(self.data))
             indptr = self.indptr * N
         return self.__class__((data, indices, indptr), shape=shape, copy=False)
+    
+    def __binsparse_descriptor__(self) -> dict:
+        """Return a `dict` equivalent to a parsed JSON [`binsparse` descriptor](https://graphblas.org/binsparse-specification/#descriptor)
+        of this array.
+
+        Returns
+        -------
+        dict
+            Parsed `binsparse` descriptor.
+        """
+        from scipy import __version__
+
+        data_dt = str(self.data.dtype)
+        if np.issubdtype(data_dt, np.complexfloating):
+            data_dt = f"complex[float{self.data.dtype.itemsize * 4}]"
+        return {
+            "binsparse": {
+                "version": "0.1",
+                "format": self.format.upper(),
+                "shape": list(self.shape),
```

Comment:
Tuple would be better here.

Reply:
JSON doesn't have tuples, and this, being a parsed JSON descriptor, would use a list. One can change https://github.com/data-apis/array-api/issues/840 and do some word-gymnastics saying "shapes always have to be tuples", but I'd rather leave this as a list to be 100% compliant.